### DISCUSSION TITLE
Create a GitHub App from a manifest

### DISCUSSION THREAD

**Discussion by giscus** [rajbos/rajbos.github.io#38] [Comments]
# Create a GitHub App from a manifest

Create a GitHub App from a manifest

https://devopsjournal.io/blog/2021/12/27/GitHub-App-from-manifest

**Comment by lvillaca** [comment]
Hi, thanks for the clear documentation! 
I've been trying to automate the creation of GitHub Apps with a similar strategy (fastapi did the trick) but found two already blockers indicated here:
- 1st I am still signing in in order to have the app registered (based on the manifest info)
- After redirection I am able to create stuff such as secrets related to the obtained app id/pem (rotation is still a pain, recreation might be a possibility)
- I would also like to automatically activate the install and use terraform to associate related repos (`github_app_installation_repositories` can be used for that, but only with a PAT - and only with an already existing installation id). Since it's been more then 2 years since this was posted, I wonder if you've found workarounds since..
BR

**Reply by rajbos** [reply]
Hi @lvillaca, I have not seen any improvements on this front, which is really a pity.
